.github/workflow/main.yml

```diff
@@ -0,0 +1,26 @@
+name: Main workflow
+on: [push]
+jobs:
+  checks:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        node: ["14.x", "16.x"]
+        os: [ubuntu-latest]
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
+      # - name: Install deps and build (with cache)
+      #   uses: bahmutov/npm-install@v1
+      - run: yarn install
+
+      - name: Next.js cache
+        uses: actions/cache@v2
```

Comment:
this btw is for vercel specifically, so we should be able to merge it independently